packages/astro/src/env/vite-plugin-env-virtual-mod.ts

```diff
@@ -127,29 +152,75 @@ function validatePublicVariables({
 	return valid;
 }
 
-function getTemplates({
+function getDts({
+	clientPublic,
+	serverPublic,
+	serverSecret,
+	fs,
+}: {
+	clientPublic: string;
+	serverPublic: string;
+	serverSecret: string;
+	fs: typeof fsMod;
+}) {
+	const template = fs.readFileSync(TYPES_TEMPLATE_URL, 'utf-8');
+
+	return template
+		.replace('// @@CLIENT@@', clientPublic)
+		.replace('// @@SERVER@@', serverPublic)
+		.replace('// @@SECRET_VALUES@@', serverSecret);
+}
+
+function getClientTemplates({
 	validatedVariables,
-	context,
 }: {
 	validatedVariables: ReturnType<typeof validatePublicVariables>;
-	context: 'server' | 'client';
 }) {
-	const contentParts: Array<string> = [];
-	const dtsParts: Array<string> = [];
+	let module = '';
+	let types = '';
 
-	for (const { key, type, value } of validatedVariables.filter((e) => e.context === context)) {
-		contentParts.push(`export const ${key} = ${JSON.stringify(value)};`);
-		dtsParts.push(`export const ${key}: ${type};`);
+	for (const { key, type, value } of validatedVariables.filter((e) => e.context === 'client')) {
+		module += `export const ${key} = ${JSON.stringify(value)};	\n`;
```

Comment:
Really small nit: Would be nice to trim out the whitespaces before the `\n` and in the other code below.